drivers/include/sht2x.h

```diff
@@ -0,0 +1,194 @@
+    SHT2X_TYPE_SHT25    = 0x03,         /**< SHT25 */
+} sht2x_type_t;
+
+/**
+ * @brief       Available resolutions
+ *
+ * @details     The values represent bits 7 and 0 in the User Register
+ */
+typedef enum {
+    SHT2X_RES_12_14BIT  = 0x00,         /**< RH=12bit, T=14bit */
+    SHT2X_RES_8_12BIT   = 0x01,         /**< RH= 8bit, T=12bit */
+    SHT2X_RES_10_13BIT  = 0x80,         /**< RH=10bit, T=13bit */
+    SHT2X_RES_11_11BIT  = 0x81,         /**< RH=11bit, T=11bit */
+} sht2x_res_t;
+
+/**
+ * @brief       Available Measuring modes
+ */
+typedef enum {
+    SHT2X_MEASURE_MODE_HOLD = 0,        /**< trigger measurement, hold master */
+    SHT2X_MEASURE_MODE_NO_HOLD = 1      /**< trigger measurement, no hold master (i.e. poll) */
+} sht2x_measure_mode_t;
+
+/**
+ * @brief       User register masks
+ * @details     Notice that the values of @c sht2x_res_t must only have the
+ *              bits in @c SHT2X_USER_RESOLUTION_MASK.
+ * @{
+ */
+#define SHT2X_USER_RESOLUTION_MASK          0x81
+#define SHT2X_USER_EOB_MASK                 0x40
+#define SHT2X_USER_HEATER_MASK              0x04
+#define SHT2X_USER_RESERVED_MASK            0x38
+#define SHT2X_USER_OTP_MASK                 0x02
+/** @} */
+
+/**
+ * @brief       Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                      /**< I2C device               */
+    uint8_t i2c_addr;                   /**< I2C address              */
+    sht2x_res_t resolution;             /**< resolution bits RH/temp  */
+    sht2x_measure_mode_t measure_mode;  /**< measurement mode used    */
+    bool is_crc_enabled;                /**< do CRC or not            */
+} sht2x_params_t;
+
+/**
+ * @brief       Device descriptor for the SHT2X sensor
+ */
+typedef struct {
+    sht2x_params_t params;              /**< Device Parameters */
+} sht2x_t;
+
+/**
+ * @brief       Initialize the given SHT2X device
+ *
+ * @param[out]  dev         Initialized device descriptor of SHT2X device
+ * @param[in]   params      The parameters for the SHT2x device
+ *
+ * @return                  SHT2X_OK on success
+ * @return                  SHT2X_ERR_I2C if given I2C is not enabled in board config
+ */
+int sht2x_init(sht2x_t* dev, const sht2x_params_t* params);
+
+/**
+ * @brief       Reset the SHT2X device
+ *
+ * @param[in]   dev         Device descriptor of SHT2X device to read from
+ *
+ * @return                  SHT2X_OK on success
+ * @return                  SHT2X_ERR_I2C if given I2C is not enabled in board config
+ */
+int sht2x_reset(sht2x_t* dev);
+
+/**
+ * @brief       Read temperature value from the given SHT2X device, returned in
+ *              centi °C
+ *
+ * @param[in]  dev          Device descriptor of SHT2X device to read from
+ *
+ * @returns                 The temperature in centi Celsius. In case of an error
+ *                          it returns INT16_MIN.
+ */
+int16_t sht2x_read_temperature(const sht2x_t* dev);
```

Comment:
If I see it correctly, this (and the other) read functions go to sleep for some time. Would you mind to add a note on that?